### PR TITLE
Show friendlier message if the API 401's

### PR DESCRIPTION
**Pull request by timabell**
### Context

Builds on https://github.com/DFE-Digital/manage-courses-api/pull/44

System was falling over when signed in user's email isn't in our mc_user list.

When the above PR is merged the API will return a 401.

### Changes proposed in this pull request

This change catches swagger exceptions across the whole site, checks for 401 and redirects the user to a new page indicating that they are signed in but that access is denied.

workspace 4_263

### CODE REVIEW

File: src/ui/Controllers/AuthController.cs [hunk 1]
```diff
@@ -50,6 +51,13 @@ public ActionResult WhoAmI()
             await HttpContext.SignOutAsync(CookieAuthenticationDefaults.AuthenticationScheme);
         }
 
+        [Authorize]
+        public ViewResult Unauthorized()
```
Comment (defong) [inline src/ui/Controllers/AuthController.cs]: Can we be a little bit more explicit with method name

ie 
There is an attribute that says you `authorise` but method name is `unauthorise`

**Comment by fofr** [comment]
I'll provide a design for the Access denied page.

File: src/ui/Controllers/AuthController.cs (hunk 1 above)
Comment (timabell) [reply src/ui/Controllers/AuthController.cs]: Good point! I'll do that...

**Comment by timabell** [comment]
Rebased.

**Pull request merged**